vizro-core/docs/pages/user-guides/custom-components.md

```diff
@@ -266,6 +266,145 @@ vm.Page.add_type("components", Jumbotron)
     [CustomComponent2]: ../../assets/user_guides/custom_components/customcomponent_2.png
 
 
+# Custom component with custom action
+
+Custom components can be used as inputs to custom actions.
+
+Following the above instructions to create a custom component, we ended with the below `OffCanvas` component.
+
+```py
+class OffCanvas(vm.VizroBaseModel):
+    type: Literal["offcanvas"] = "offcanvas"
+    title: str
+    content: str
+
+    def build(self):
+        return html.Div(
+            [
+                dbc.Offcanvas(
+                    children=html.P(self.content),
+                    id=self.id,
+                    title=self.title,
+                    is_open=False,
+                ),
+            ]
+        )
+```
+To enable our custom component to work with our custom action, we need to add few additional lines of code.
+
+1. First, you have to add `actions` argument to your custom component. Type of `actions` argument is `List[Action]`.
+2. Next step is to define `_input_property` of the component. `_input_property` is the property that is used as the input to the custom action callback
+   ```py
+    _input_property: str = PrivateAttr("is_open")
+   ```
+3. Next, you need to `_set_actions`. By setting action, change in 'is_open' property of our custom component will trigger the action.
+   ```py
+    _set_actions = _action_validator_factory("is_open")
+   ```
```

Comment:
None of these are required to enable a custom component (property) to be the `input`/`output` of a custom action.

Some of these requirements (`1.` and `3.`) are really needed if the custom component `triggers` a custom action. (`2.` isn't required even then, because `_input_property` only stands for component's `property` which will be used if the component is used as the filter/parameter selector).

However, I'd really like to see another example below that describes how to configure a custom component to `trigger` an action (custom or predefined). Then you can add requirements (`1.` and `3.`) and an example. 
In that case, the title `"## Custom component with custom action"` could be renamed to something like `"## Custom component with actions"` which could contain these two subtitles:
 - "### Custom component as input/output of custom action"
 - "### Triggering actions with a custom component"

Reply:
Sorry, I didn't see this comment above before I made the edits. I've just edited for style to bold the first sentence of each instruction and make it imperative, so it's clearer for people to swiftly scan the list. If you make changes for Petar's comments, please just follow that guideline.

Reply:
Removed this section for offcanvas component. Added another example for when the custom component triggers the action.